plugins/woocommerce-admin/client/core-profiler/index.tsx

```diff
@@ -300,25 +347,59 @@ const coreProfilerStateMachineDefinition = createMachine( {
 				component: BusinessInfo,
 			},
 		},
+		preSkipFlowBusinessLocation: {
+			invoke: [
+				{
+					src: 'getCountries',
+					onDone: [
+						{
+							actions: [ 'handleCountries' ],
+							target: 'skipFlowBusinessLocation',
+						},
+					],
+					onError: {
+						target: 'skipFlowBusinessLocation',
+					},
+				},
+			],
+		},
 		skipFlowBusinessLocation: {
 			on: {
 				BUSINESS_LOCATION_COMPLETED: {
-					target: 'settingUpStore',
+					target: 'postSkipFlowBusinessLocation',
 					actions: [
 						assign( {
 							businessInfo: (
```

Comment:
Looks like `settingUpStore` is an empty state. Could we make it to a function so we can invoke it instead?

Reply:
settingUpStore is the placeholder state for the regular non-skip flow actually, I should have written a comment about that sorry!

I think we can just add it as an exit action on 'skipFlowBusinessLocation' perhaps? 
Something like 'updateTrackingOption()' without the extra explat stuff

Reply:
exit action sounds good 👍 I'll test it out soon

Reply:
@rjchow 

Updated in https://github.com/woocommerce/woocommerce/pull/38019/commits/c6cc926aa7396b199380a30c81ca678d92b88d6b

I used a new state to invoke the [UpdateBusinessLocation](https://github.com/woocommerce/woocommerce/pull/38019/commits/c6cc926aa7396b199380a30c81ca678d92b88d6b#diff-2944845107cf67e4f7c24f66369c78ea657b27b8816abcb406d26f4991304ee2R200) function instead of ㄴusing an action with exit. It seems that actions must be pure functions without side effects according to https://github.com/statelyai/xstate/issues/268

Reply:
Looks good! Checked and it works correctly for me